### PR TITLE
Add internal router and trainer

### PR DESCRIPTION
**Pull request by arcticfly**
This PR allows our training docker containers to get a shared access signature url from the main server to download a fine-tuned model's training data from Azure.

### Changes
* Add `BaseModel` enum, `~utils/baseModel.ts`
* Add `codegen:internalClient` pnpm command
* Add `api/internal/v1` router with `getTrainingInfo` route
* Add docker client with basic test
* Add `generateBlobDownloadUrl` function

### CODE REVIEW

File: app/.env.example [hunk 1]
```diff
@@ -13,6 +13,9 @@
 # https://www.prisma.io/docs/reference/database-reference/connection-urls#env
 DATABASE_URL="postgresql://postgres:postgres@localhost:5432/openpipe?schema=public"
 
+# Used by docker containers to request training data from the API
+DOCKER_SECRET="placeholder"
```
Comment (corbt) [inline app/.env.example]: This isn't a great name. For one thing, all our software runs in Docker. Maybe AUTHENTICATED_SYSTEM_KEY?

File: app/src/pages/api/internal/v1/openapi.json.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+import { type NextApiRequest, type NextApiResponse } from "next";
+import { generateOpenApiDocument } from "trpc-openapi";
+import { v1ApiRouter } from "~/server/api/internal/v1Api.router";
+
+export const openApiDocument = generateOpenApiDocument(v1ApiRouter, {
+  title: "OpenPipe Docker API",
```
Comment (corbt) [inline app/src/pages/api/internal/v1/openapi.json.ts]: "OpenPipe Internal API". Docker isn't a useful description. We may or may not use Docker as the runtime, and anyway, our main app runs in Docker too, so it's a weird thing to use as a distinguisher.

File: app/src/server/api/internal/openApiTrpc.ts [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+          ...shape.data,
+          zodError: error.cause instanceof ZodError ? error.cause.flatten() : null,
+        },
+      };
+    },
+  });
+
+export const createOpenApiRouter = t.router;
+
+export const openApiPublicProc = t.procedure;
```
Comment (corbt) [inline app/src/server/api/internal/openApiTrpc.ts]: Shouldn't this be `openApiProtectedProc`?

File: docker-client/openapi.json [hunk 4]
```diff
@@ -0,0 +1,118 @@
+{
```
Comment (corbt) [inline docker-client/openapi.json]: I don't see any code to generate this file, did you just manually copy/paste it? If we aren't autogenerating this file probably better not to have it at all and run `poetry run openapi-python-client generate --url http://localhost:3000/api/internal/openapi.json` or whatever.

**Review by corbt — CHANGES_REQUESTED** [review]

File: docker-client/openapi.json (hunk 4 above)
Comment (corbt) [reply docker-client/openapi.json]: Nvm I see where you generate this file now.

**Comment by arcticfly** [comment]
Updated the PR to use the word "Docker" in fewer places. Not sure why, but I was really focused on running this job in a docker container when I devised the folder structure and variable names!

**Review by corbt — APPROVED** [review]

**Pull request merged**